[ADD] hr_timesheet_begin_end_break: Add break duration field.

Slivi-sliv opened the pull request:
**Description**
This PR extends the functionality of the _hr_timesheet_begin_end_ module by adding a break_duration field to timesheet lines. It allows users to record their start and end times while automatically deducting a break (e.g., lunch) from the total hours.

**Changes**
- **Module Dependency**: Built as an extension to hr_timesheet_begin_end.
- **Data Model**: Added break_duration field (float_time) to account.analytic.line.
- **Logic Enhancement**: * Overridden _check_time_start_stop constraint to validate that unit_amount equals the difference between start/end times minus the break.
    - Updated onchange_hours_start_stop to dynamically calculate duration in the UI when the break is entered.
- **UI/UX**: Added the break_duration field to the Timesheet tree and form views, positioned next to the start/end times for better flow.
- **Testing**: Added new test cases to cover duration calculations with breaks and updated existing constraints to prevent conflicts with the base module.

Slivi-sliv committed:
[IMP] hr_timesheet_begin_end: Add break duration field.

Added a break_duration field that is subtracted from the total duration (unit_amount).

Slivi-sliv committed:
style: apply pre-commit fixes

Slivi-sliv committed:
style: fix end of file and whitespace

Slivi-sliv committed:
tyle: fix end of file and whitespace

Slivi-sliv committed:
Pvza to Brake

Slivi-sliv committed:
imp: add break duration to timesheets and tasks

This PR introduces the possibility to record break durations in timesheets.
When using begin/end hours, the break duration is now subtracted from the total time to calculate the correct unit amount.

### Changes:
- **Models**: Added `break_duration` field to `account.analytic.line`.
- **Logic**: Updated `onchange` and `constrains` to include break duration in the total time calculation:
  `Unit Amount = (End Hour - Begin Hour) - Break Duration`.
- **Views**:
  - Added the Break field to the main Timesheet tree and form views.
  - Added the Break field to the Timesheets sub-view within Project Tasks.
- **Manifest**: Added `project` to dependencies to ensure proper view inheritance in tasks.

### Practical Example:
- Begin: 08:00
- End: 16:30
- Break: 00:30
- **Total Unit Amount**: 8.00 hours

Slivi-sliv committed:
Update __manifest__.py

Slivi-sliv committed:
Update test_timesheet_begin_end.py

Slivi-sliv committed:
Update test_timesheet_begin_end.py

Slivi-sliv committed:
Update test_timesheet_begin_end.py

Slivi-sliv committed:
Update test_timesheet_begin_end.py

Slivi-sliv committed:
Update test_timesheet_begin_end.py

CRogos commented:
fyi:
@Slivi-sliv I was also using hr_timesheet_begin_end because I wasn't aware of https://github.com/OCA/project/pull/1588

I also build a begin_end module for this module https://github.com/OCA/timesheet/pull/833

Not sure if you should implement this into a new module, to prevent this field for users who do not need them?

Slivi-sliv committed:
Update test_timesheet_begin_end.py

Slivi-sliv committed:
Update test_timesheet_begin_end.py

Slivi-sliv commented:
@CRogos Thank you for the suggestion.
I agree that modularity is important. I have moved the break duration functionality into a new module named hr_timesheet_begin_end_break within this same PR.
This way, the feature remains optional. I've also reverted the changes in the base module.

Slivi-sliv committed:
Move break functionality to a separate module

Slivi-sliv committed:
Fix manifest, linting and add missing README

Adjusted manifest authors, added website, fixed redundancy in models and added README.rst to comply with OCA standards.

Slivi-sliv committed:
Add hr_timesheet_begin_end_break module

Slivi-sliv committed:
docs: fix formatting and linting issues

Slivi-sliv committed:
Fix translation formatting (W8120) and apply linting

Slivi-sliv committed:
Fix translation formatting

Slivi-sliv committed:
Fix translation formatting

Slivi-sliv committed:
Add comment to manifest

Slivi-sliv committed:
style: apply automated formatting fixes

Slivi-sliv committed:
style: add missing newline at end of files

Slivi-sliv committed:
fix: validation logic, test formatting and RST syntax

Slivi-sliv committed:
fix: linting issues

Slivi-sliv committed:
style: final ruff formatting

Slivi-sliv committed:
fix: bypass parent constraint to allow break duration

Slivi-sliv committed:
code fix

Slivi-sliv committed:
fix: update tests for break duration

Slivi-sliv committed:
fix code

Slivi-sliv committed:
refactor: improve code formatting according to OCA standards

Slivi-sliv committed:
style: fix whitespace and indentation

bosd commented:
@Slivi-sliv  Great Addition!

Few remarks:
On a project, with a time record (4 hours) , enter the break time. e.g. 0:30 min.
What happens: The time record entry gets set to 0 hours. And only registering the break time.
Expected result: Substract the break from the recorded time. 
0:30 Break
3:30 Project time


:bulb: Maybe add a :coffee:  button to register break time

Can you squash the commits to get this one ready for merge?

alexey-pelykh commented on hr_timesheet_begin_end_break/models/account_analytic_line.py:
```diff
@@ -0,0 +1,71 @@
+        # does not account for break_duration and would raise a ValidationError.
+
+        for line in self:
+            if not line.time_start and not line.time_stop:
```
Skipping super entirely and duplicating the overlap check from the base module is fragile. If `hr_timesheet_begin_end` updates its constraint (new checks, bug fixes), this module won't get those changes. Consider calling super in a try/except and only overriding the duration mismatch error, or refactoring the base constraint to be more extensible.

alexey-pelykh reviewed (commented):
Nice addition -- break duration is a common request for begin/end time tracking.

A few observations:
- `_check_time_start_stop` skips `super()` entirely and reimplements the full constraint including the overlap check. This means any future changes to the base constraint in `hr_timesheet_begin_end` won't propagate here. Safer to isolate only the duration validation override and let super handle overlap/ordering.
- Some test comments are in Slovenian (`Zaokroženo na 2 decimalki...`) -- should be English for OCA.
- README.rst is minimal (5 lines) -- the OCA fragments generator would fill this out nicely.

bosd commented:
@Slivi-sliv Can you please attend to the reviews. Would like to have this one on v19
